ledfx/effects/gifplayer.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import os
+import time
+import urllib.request
+
+import voluptuous as vol
+from PIL import Image
+
+from ledfx.consts import LEDFX_ASSETS_PATH
+from ledfx.effects.twod import Twod
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class GifPlayer(Twod):
+    NAME = "GIF Player"
+    CATEGORY = "Matrix"
+    HIDDEN_KEYS = Twod.HIDDEN_KEYS + []
+    ADVANCED_KEYS = Twod.ADVANCED_KEYS + ["bilinear", "blur"]
+    DEFAULT_GIF_PATH = f"{os.path.join(LEDFX_ASSETS_PATH, 'animated.gif')}"
+
+    CONFIG_SCHEMA = vol.Schema(
+        {
+            vol.Optional(
+                "bilinear",
+                description="default NEAREST, use BILINEAR for smoother scaling, expensive on runtime takes a few ms",
+                default=False,
+            ): bool,
+            vol.Optional(
+                "url source", description="Load image from", default=""
+            ): str,
+            vol.Optional(
+                "bounce",
+                description="Bounce the GIF instead of looping",
+                default=False,
+            ): bool,
+            vol.Optional(
+                "GIF FPS", description="How fast to play the gif", default=10
+            ): vol.All(vol.Coerce(int), vol.Range(min=1, max=60)),
+        }
+    )
+
+    def __init__(self, ledfx, config):
+        super().__init__(ledfx, config)
+
+    def config_updated(self, config):
+        super().config_updated(config)
+        self.gif_fps = self._config["GIF FPS"]
+        self.bounce = self._config["bounce"]
+        self.resize = (
+            Image.BILINEAR if self._config["bilinear"] else Image.NEAREST
+        )
+        self.frames = []
+        self.current_frame = 0
+        self.start_time = time.time()
+        self.last_frame_time = self.start_time
+        self.init = True
+
+    def audio_data_updated(self, data):
+        # Non-reactive by design - do nothing
+        return
+
+    def open_image(self, image_path):
+        try:
+            if image_path.startswith("http://") or image_path.startswith(
+                "https://" or image_path.startswith("file://")
+            ):
+                with urllib.request.urlopen(image_path) as url:
+                    return Image.open(url)
+            else:
+                return Image.open(image_path)  # Directly open for local files
+        except Exception as e:
+            _LOGGER.error("Failed to open image, using default GIF: %s", e)
+            return Image.open(self.DEFAULT_GIF_PATH)
+
+    def do_once(self):
+        super().do_once()
+
+        url_path = self._config["url source"]
+
+        if url_path == "":
+            self.gif = self.open_image(self.DEFAULT_GIF_PATH)
+        else:
+            self.gif = self.open_image(url_path)
+        self.process_gif()
+        # Seed the frame data with the first frame
+        self.frame_data = self.frames[0]
+        self.gif_frame_duration = 1 / self.gif_fps
+        self.init = False
+
+    def draw(self):
+        if self.init:
```

Comment:
You don't need this call in draw, it will be done for you by the code in twod method render(...) before call gets called, so self.init will never be true when you get here.